src/config.c

```diff
@@ -1848,7 +1848,7 @@ static sds sdsConfigGet(standardConfig *config) {
 static void sdsConfigRewrite(standardConfig *config, const char *name, struct rewriteConfigState *state) {
     sds val = config->flags & MODULE_CONFIG ? getModuleStringConfig(config->privdata) : *config->data.sds.config;
     rewriteConfigSdsOption(state, name, val, config->data.sds.default_value);
-    if (val) sdsfree(val);
+    if ((val) && (config->flags & MODULE_CONFIG)) sdsfree(val);
```

Comment:
looking at this fix, and the rest of the code, don't we have the same problem in `sdsConfigSet`?
```c
    if (new != prev && (new == NULL || prev == NULL || sdscmp(prev, new))) {
        sdsfree(prev);
```

Reply:
Doesn't look like. 

Right after the first sdsfree(), we set new value to `*config->data.sds.config`.
As for the second sdsfree(), we apply it only in case of module, which for sure a temporary sds that can be deleted.

Reply:
ohh, ok.
so we get the prev value, which for modules is a temporary allocation, and for non-modules is the actual config.
if we succeeded to change the config, we need to release the prev either way (calling setModuleStringConfig will cause the module to release the actual config too).
and if we failed to change the config, we need to release the prev only in case it's a module (since that's a temp allocation).
is that right?

the whole thing smells, and could at least benefit from some additional comments...
@moticless can you please do that? (i can't merge my own PRs)